processor/src/host/default.rs

```diff
@@ -16,6 +19,7 @@ pub struct DefaultHost<D: DebugHandler = DefaultDebugHandler> {
     store: MemMastForestStore,
     event_handlers: EventHandlerRegistry,
     debug_handler: D,
+    source_manager: Arc<dyn SourceManagerSync>,
```

Comment:
Question: is there a real need to use `Arc<dyn SourceManagerSync>` in the `DefaultHost`? An alternative would be to just use `DefaultSourceManager`. This would simplify things, and I'm not sure we'd be giving up a lot (i.e. the `DefaultHost` is already pretty opinionated.

If we did want to make it more flexible, another option would be to follow the same pattern as with the `DebugHandler` - i.e., define the `DefaultHost` as follows:
```rust
pub struct DefaultHost<D: DebugHandler = DefaultDebugHandler, S: SourceManager = DefaultSourceManager> {
    store: MemMastForestStore,
    event_handlers: EventHandlerRegistry,
    debug_handler: D,
    source_manager: S,
}
```